frontend/src/components/DietPlan/DietPlan.tsx

```diff
@@ -27,45 +46,75 @@ export default function DietPlan() {
     >
       <ImageBackground source={logoBlack} className="w-screen h-[30vh]" />
 
-      <View className="flex-row justify-center gap-9 pb-6">
-        <TouchableOpacity onPress={() => setUiView(UI_TYPES.PROTEIN)}>
-          <View className="bg-emerald-300 p-2 rounded">
-            <Text className="text-lg font-bold">חלבונים</Text>
-          </View>
-        </TouchableOpacity>
-        <TouchableOpacity onPress={() => setUiView(UI_TYPES.CARBS)}>
-          <View className="bg-emerald-300 p-2 rounded">
-            <Text className="text-lg font-bold">פחמימות</Text>
-          </View>
-        </TouchableOpacity>
-      </View>
-      {meals.map((meal) => (
-        <View className="p-4 rtl items-center" key={meal.title}>
-          <Text className="text-emerald-300 underline font-bold text-lg p-2">{meal.title}</Text>
-          <View className="flex-row">
-            {typeof meal.optOne == `string` ? (
-              <Text className="text-white">{meal.optOne}</Text>
-            ) : (
-              <View className="flex-row-reverse w-[100vw] flex-wrap p-1 justify-center">
-                {meal.optOne.map((opt) => (
-                  <Text className="text-white px-1">{opt} /</Text>
-                ))}
-              </View>
-            )}
-          </View>
-          <Text className="text-emerald-300 font-bold">+</Text>
-          <View className="flex-row-reverse gap-2">
-            <Text className="text-white py-1">{meal.optTwo}</Text>
-            <Text className="text-emerald-300 font-bold py-1">+</Text>
-            <Text className="text-white py-1">{meal.optThree}</Text>
+      {dietPlan?.meals.map((meal,i) => (
+        <View key={i} style={styles.meal}>
+          <View style={styles.mealIcon}>
+            <NativeIcon library="MaterialCommunityIcons" name="food-outline" color={Colors.light} size={20}/>
+          <Text style={styles.mealTitle}>ארוחה {i+1}</Text>
           </View>
+          <MealContainer  meal={meal}/>
         </View>
       ))}
+    
+      <MenuItemModal foodGroup={selectedFoodGroup} isOpen={isModalOpen} dismiss={()=>setIsModalOpen(false)}/>
+      <FABGroup
+        style={{bottom:100}}
+        open={isFabOpen}
+        visible
+        icon={isFabOpen?`close`:`food-outline`}
+        actions={[
+          {
+              icon: 'fish',
+              label: 'חלבונים',
+              onPress: () => openModal(`protein`),
+            },
+            {
+              icon: 'baguette',
+              label: 'פחמימות',
+              onPress: () => openModal(`carbs`),
+            },
+            {
+              icon: 'cheese',
+              label: 'שומנים',
+              onPress: () => openModal(`fats`),
+            },
+            {
+            icon:`leaf`,
+            label:`ירקות`,
+            onPress:()=> openModal(`vegetables`)   
+          }
+        ]}
+        onStateChange={()=>setIsFabOpen(!isFabOpen)}
```

Comment:
onStateChange has props log them to see what they are and see if you can use them. Also this is not the proper way to update state